addon/mixins/disposable.js

```diff
@@ -10,11 +10,7 @@ import getOrAllocate from '../utils/get-or-allocate';
  @public
  */
 export default Mixin.create({
-  init() {
-    this._super(...arguments);
-
-    this._registeredDisposables = undefined;
-  },
+  [WILL_DESTROY_PATCHED]: true,
```

Comment:
this doesn't seem used, lets remove.

Reply:
It's used in the assertion. https://github.com/rwjblue/ember-lifeline/pull/63/files#diff-69582adbea9112eef4525140eef26a3bR38

Reply:
OK, so to be specific here, I believe we still need and want this assertion. The intent of this library is to ensure that we're registering work, and correctly disposing of said work when an object is 'destroyed'. The property as symbol approach is aimed at ensuring that we have correctly implemented the call to `runDisposables(this`, which currently is the way we'll be disposing of work.

This property isn't so much intended to be used to ensure the use of the mixin, but instead used as a way to signal to the library that we've correctly implemented `runDisposables` in the destruction of the object.

I hope this makes more sense.